src/ngraph_encapsulate_op.cc

```diff
@@ -212,6 +213,26 @@ class NGraphEncapsulateOp : public OpKernel {
     return Status::OK();
   }
 
+  void mem_usage(double& vm_usage, double& resident_set) {
+    vm_usage = 0.0;
+    resident_set = 0.0;
+
+    // the two fields we want
+    unsigned long vsize;
+    long rss;
+    {
+      std::string ignore;
+      std::ifstream ifs("/proc/self/stat", std::ios_base::in);
+      ifs >> ignore >> ignore >> ignore >> ignore >> ignore >> ignore >> ignore >> ignore >> ignore >> ignore
+      >> ignore >> ignore >> ignore >> ignore >> ignore >> ignore >> ignore >> ignore >> ignore >> ignore
+      >> ignore >> ignore >> vsize >> rss;
```

Comment:
this is a fun way of parsing the file :-D

We could use `ng.split` (https://github.com/NervanaSystems/ngraph/blob/master/src/ngraph/util.hpp#L79)

Reply:
Fixed the way of file parsing